src/lifecycleComponent/Mount/Mount.js

```diff
@@ -0,0 +1,43 @@
+import { Component } from 'react';
+
+import './mount.scss';
+
+export class Mount extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      status: 'init',
+      colorBg: '#ccc'
+    };
+  }
+
+  updateStatus = (e) => {
+    this.setState({
+      status: `${this.state.status}+`
+    });
+
+    e.preventDefault();
+  }
+
+  componentWillReceiveProps({ code }) {
+    this.setState({ colorBg: '#ccc' });
+    if (code % 3 === 0) this.setState({ colorBg: 'green' });
+    if (code % 5 === 0) this.setState({ colorBg: 'blue' });
+    if (code % 3 === 0 && code % 5 === 0) this.setState({ colorBg: '#0ff' });
+    if (code === 0) this.setState({ colorBg: '#ccc' });
```

Comment:
it's better to call setState() after conditions, because in case _(code % 3 === 0 && code % 5 === 0)_  you'll have two extra calls